compose-builder/tui-generator.sh

```diff
@@ -0,0 +1,299 @@
+    [mqtt-bus]="Configure MQTT Message Bus"
+    [zmq-bus]="Configure ZMQ Message Bus"
+)
+
+
+####################################################################
+#    Additional Services Options Display Function                  #
+####################################################################
+function additionalServiceOption() {
+    message="Some Additional Services are available that can also be included. \n
+            Press <SPACEBAR> to Select \n
+            Press <ENTER> to Skip
+            "
+
+    # Generate a Specific form of Array required by whiptail checklist
+    # FORMAT: "<INDEX> <DESCRIPTION> <OFF>"
+    arglist=()
+    for index in "${additionalOpts[@]}";
+    do
+        arglist+=("$index")
+        arglist+=("${additionalOptsDesc[$index]}")
+        arglist+=("OFF") # Nothing is selected by-default
+    done
+    SELECTED_OTHERS+=$($WHIPTAIL --title "App Services" \
+                --notags --separate-output \
+                --ok-button Next \
+                --nocancel \
+                --checklist "$message" $LINES $COLUMNS $(( $LINES - 12 )) \
+                -- "${arglist[@]})" \
+                3>&1 1>&2 2>&3)
+}
+
+
+####################################################################
+#    App Services Options Display Function                         #
+####################################################################
+function appServiceOption() {
+    message="Available App Services to include in your compose file. \n
+            Press <SPACEBAR> to Select \n
+            Press <ENTER> to Skip
+            "
+
+    # Generate a Specific form of Array required by whiptail checklist
+    # FORMAT: "<INDEX> <DESCRIPTION> <OFF>"
+    arglist=()
+    for index in "${appServices[@]}";
+    do
+        arglist+=("$index")
+        arglist+=("${appServiceDesc[$index]}")
+        arglist+=("OFF") # Nothing is selected by-default
+    done
+    SELECTED_APPSERVICES+=$($WHIPTAIL --title "App Services" \
+                --ok-button Next \
+                --nocancel \
+                --notags --separate-output \
+                --checklist "$message" $LINES $COLUMNS $(( $LINES - 12 )) \
+                -- "${arglist[@]}" \
+                3>&1 1>&2 2>&3)
+}
+
+
+
+####################################################################
+#    ARM64 Question Display Function                               #
+####################################################################
+function displayArm64() {
+    message="Would you like to use ARM64 Images to generate the Compose file?"
+    $WHIPTAIL --title "Images Architecture" --yesno --defaultno "$message" $LINES $COLUMNS
+}
+
+
+####################################################################
+#    No-Security Question Display Function                         #
+####################################################################
+function displayNoSecty() {
+    message="Would you like to generate a non-secure configuration of the Compose file?"
```

Comment:
I feel this question is a little backwards, or worded stangely. Maybe instead of yes-no it should be `Secure` and `Non-Secure` for the buttons kind of like the final step?

Reply:
@ajcasagrande How about refactoring the question to

> Choose a Security configuration for EdgeX?

where the options would be either `Secure` or `Non-Secure`